client/landing/jetpack-cloud/sections/backups/rewind-flow/restore.tsx

```diff
@@ -1,15 +1,206 @@
+import { useTranslate } from 'i18n-calypso';
+import React, { FunctionComponent, useCallback, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '@automattic/components';
+import { defaultRewindConfig, RewindConfig } from './types';
+import { rewindRestore } from 'state/activity-log/actions';
+import { useLocalizedMoment } from 'components/localized-moment';
+import CheckYourEmail from './rewind-flow-notice/check-your-email';
+import getInProgressRewindPercentComplete from 'state/selectors/get-in-progress-rewind-percent-complete';
+import getInProgressRewindStatus from 'state/selectors/get-in-progress-rewind-status';
+import getRewindState from 'state/selectors/get-rewind-state';
+import getSiteUrl from 'state/selectors/get-site-url';
+import Gridicon from 'components/gridicon';
+import ProgressBar from './progress-bar';
+import QueryRewindState from 'components/data/query-rewind-state';
+import RewindConfigEditor from './rewind-config-editor';
+import RewindFlowNotice, { RewindFlowNoticeLevel } from './rewind-flow-notice';
+import Spinner from 'components/spinner';
 
 interface Props {
 	rewindId: string;
 	siteId: number;
 }
 
-const BackupRestoreFlow: FunctionComponent< Props > = () => {
-	return <div>Restore</div>;
+//todo: move to dedicated types file
+interface RewindState {
+	state: string;
+	rewind?: {
+		status: 'queued' | 'running' | 'finished' | 'fail';
+	};
+}
+
+const BackupRestoreFlow: FunctionComponent< Props > = ( { rewindId, siteId } ) => {
+	const dispatch = useDispatch();
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+
+	const [ rewindConfig, setRewindConfig ] = useState< RewindConfig >( defaultRewindConfig );
+	const [ userHasRequestedRestore, setUserHasRequestedRestore ] = useState< boolean >( false );
+
+	const siteUrl = useSelector( state => getSiteUrl( state, siteId ) );
+	const rewindState = useSelector( state => getRewindState( state, siteId ) ) as RewindState;
+
+	const loading = rewindState.state === 'uninitialized';
+
+	const restoreTimestamp = moment.unix( rewindId ).format( 'LLL' );
+
+	const inProgressRewindStatus = useSelector( state =>
+		getInProgressRewindStatus( state, siteId, rewindId )
+	);
+	const inProgressRewindPercentComplete = useSelector( state =>
+		getInProgressRewindPercentComplete( state, siteId, rewindId )
+	);
+
+	const requestRestore = useCallback(
+		() => dispatch( rewindRestore( siteId, rewindId, rewindConfig ) ),
+		[ dispatch, rewindConfig, rewindId, siteId ]
+	);
+
+	const onConfirm = () => {
+		setUserHasRequestedRestore( true );
+		requestRestore();
+	};
+
+	const renderConfirm = () => (
+		<>
+			<div className="rewind-flow__header">
+				<Gridicon icon="history" size={ 48 } />
+			</div>
+			<h3 className="rewind-flow__title">{ translate( 'Restore site' ) }</h3>
+			<p className="rewind-flow__info">
+				{ translate(
+					'{{strong}}%(restoreTimestamp)s{{/strong}} is the selected point for your restore. ',
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( '{{strong}}%(restoreTimestamp)s{{/strong}} is the selected point for your restore' )` *ES Score: 9*